src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -977,6 +978,7 @@ public ListBoxModel doFillFleetItems(@QueryParameter final boolean showAllFleets
                                              @QueryParameter final String awsCredentialsId,
                                              @QueryParameter final String fleet) {
             final ListBoxModel model = new ListBoxModel();
+            model.add(0, new Option("- please select -", "empty"));
```

Comment:
Can we just use "" instead of "empty"? Will be easier to validate.

Wasn't there a code change for "Set default for asg to please-select and tried to save. Verified a failure occurs on saving the ec2 fleet could configuration"? I guess that depends on how we want to communicate the failure to the user. If you can attach a screenshot of what the failed save looks like, that will be helpful.
I was thinking of a new validation check like doCheckFleet to fail fast.

Reply:
Not too familiar with jelly yet but client-side validation to make this a required field (along with `doCheckFleet`) will be a good start I think.

Reply:
We should prob make this default option `selected` ?
```
new Option("- please select -", "empty", true)
```

Reply:
> Can we just use "" instead of "empty"? Will be easier to validate.

Sure

> I guess that depends on how we want to communicate the failure to the user. If you can attach a screenshot of what the failed save looks like, that will be helpful.
I was thinking of a new validation check like doCheckFleet to fail fast.

Previously, there was a generic exception thrown when the default fleet was chosen. I've updated to use the `doCheckFleet` method to validate the selected fleet. Please, see screenshot for the error provided. 

<img width="1458" alt="Screen Shot 2023-07-12 at 3 21 38 PM" src="https://github.com/jenkinsci/ec2-fleet-plugin/assets/35543771/4eff6a09-bcb4-457b-a5fe-fcab469a8085">

> We should prob make this default option selected ?

By default, the first item in the `select` list is chosen if this is a new fleet or a previous asg group is removed. I agree it would not hurt to explicitly set the default as the selected option though.